### PR TITLE
Use remote action

### PR DESCRIPTION
**Pull request by Hoverbear**
Resulting from https://github.com/DeterminateSystems/nix-installer/pull/179#discussion_r1068543591, we split off the action to https://github.com/DeterminateSystems/nix-installer-action

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,10 +11,10 @@ jobs:
     runs-on: ubuntu-22.04
     steps:
       - uses: actions/checkout@v3
-      - uses: cachix/install-nix-action@v17
+      - name: Install Nix
+        uses: determinatesystems/nix-installer-action@main
```
Comment (cole-h) [inline .github/workflows/ci.yml:15]:
```suggestion
        uses: DeterminateSystems/nix-installer-action@main
```

(and everywhere else)

File: README.md [hunk 2]
```diff
@@ -129,7 +129,7 @@ You can remove a `nix-installer`-installed Nix by running
 
 ## As a Github Action
 
-You can use `nix-installer` as a Github action like so:
+You can use the [`nix-installer-action`](https://github.com/DeterminateSystems/nix-installer-action) Github action like so:
```
Comment (cole-h) [inline README.md:132]: nit:

```suggestion
You can use the [`nix-installer-action`](https://github.com/DeterminateSystems/nix-installer-action) GitHub Action like so:
```

File: README.md [hunk 3]
```diff
@@ -144,7 +144,7 @@ jobs:
     steps:
     - uses: actions/checkout@v3
     - name: Install Nix
-      uses: DeterminateSystems/nix-installer@main
+      uses: DeterminateSystems/nix-installer-action
```
Comment (cole-h) [inline README.md:147]: I think it's required to specify a version / ref here...? Maybe?

Comment (Hoverbear) [reply README.md:147]: I don't think it is! :grimacing:

Comment (cole-h) [reply README.md:147]: :scream:

**Review by cole-h — APPROVED** [review]

**Pull request merged**